### PR TITLE
[IMP] account_payment_order: extensibility of _prepare_move

### PR DESCRIPTION
**Pull request by sbidoul**
Pass the list of bank lines to _prepare_move so it is possible to customize the move (eg it's name) based on the lines being paid.

### CODE REVIEW

File: account_payment_order/models/account_payment_order.py [hunk 1]
```diff
@@ -330,7 +330,7 @@ def generated2uploaded(self):
         return True
 
     @api.multi
-    def _prepare_move(self):
+    def _prepare_move(self, bank_lines=None):
```
Comment (pedrobaeza) [inline account_payment_order/models/account_payment_order.py:333]: Where is bank_lines used?

Comment (sbidoul) [inline account_payment_order/models/account_payment_order.py:333]: It's not used in this module, but it is useful when overriding `_prepare_move' so some parts of the move can be customized in extension modules.

Comment (pedrobaeza) [inline account_payment_order/models/account_payment_order.py:333]: Then is the purpose to change bank payment lines? Then that should be done in another method I think

Comment (sbidoul) [inline account_payment_order/models/account_payment_order.py:333]: No, I don't want to change the bank payment lines. I want to compute `ref` from the bank payment lines.

Comment (pedrobaeza) [inline account_payment_order/models/account_payment_order.py:333]: OK, understood, thanks.

**Comment by pedrobaeza** [comment]
:+1:

**Comment by alexis-via** [comment]
ok, why not :+1:

**Comment by sbidoul** [comment]
@alexis-via in scenarios where we don't group payment lines, it's good to generate more precise refs and labels in account moves.

**Comment by pedrobaeza** [comment]
Merging as harmless and a default value has been provided.

**Pull request merged**